libs/ngxtension/computed-async/src/computed-async.ts

```diff
@@ -72,46 +96,157 @@ interface ComputedAsyncOptions<T> extends CreateComputedOptions<T> {
+	options: {
+		initialValue?: undefined;
+		requireSync?: false;
+	} & ComputedAsyncOptions<T>,
+): Signal<T | undefined>;
+
+// Options with `null` initial value -> `null`.
 export function computedAsync<T>(
-	computation: (previousValue?: T | undefined) => ComputationResult<T>,
-  options: {initialValue:undefined} & ComputedAsyncOptions<T>,
-): Signal<T|undefined>
+	computation: ObservableComputation<T>,
+	options: {
+		initialValue?: null;
+		requireSync?: false;
+	} & ComputedAsyncOptions<T>,
+): Signal<T | null>;
+
+// Options with `undefined` initial value and `requireSync` -> strict result type.
 export function computedAsync<T>(
-	computation: (previousValue?: T | undefined) => ComputationResult<T>,
-	options: ComputedAsyncOptions<T>,
-): Signal<T>
+	computation: ObservableComputation<T>,
+	options: OptionsWithRequireSync<T> & { initialValue?: undefined },
+): Signal<T>;
+
+// Options with `T` initial value and `requireSync` -> strict result type.
 export function computedAsync<T>(
-	computation: (previousValue?: T | undefined) => ComputationResult<T>,
-	options?: ComputedAsyncOptions<T>,
-): Signal<T|undefined> {
+	computation: ObservableComputation<T>,
+	options: OptionsWithRequireSync<T> & { initialValue: T },
+): Signal<T>;
+
+// Options with `T` initial value and no `requireSync` -> strict result type.
+export function computedAsync<T>(
+	computation: ObservableComputation<T>,
+	options: OptionsWithInitialValue<T>,
+): Signal<T>;
+
+export function computedAsync<T>(
+	computation: (
+		previousValue?: T | undefined,
+	) => Promise<T> | Observable<T> | T | undefined,
+	options?: any,
+): Signal<T> {
 	return assertInjector(computedAsync, options?.injector, () => {
 		const destroyRef = inject(DestroyRef);
 
 		// source$ is a Subject that will emit the new source value
 		const sourceEvent$ = new Subject<Promise<T> | Observable<T>>();
 
-		// will hold the current value
-		const sourceValue = signal<T | undefined>(
-			options?.initialValue ?? undefined,
-		);
+		// sourceValue is a signal that will hold the current value and the state of the value
+		let sourceValue: WritableSignal<State<T>>;
+
+		if (options?.requireSync && options?.initialValue === undefined) {
+			const initialComputation = computation(undefined);
+
+			// we don't support promises with requireSync and no initialValue
+			if (isPromise(initialComputation)) {
+				throw new Error(REQUIRE_SYNC_PROMISE_MESSAGE);
+			}
+
+			sourceValue = signal<State<T>>({ kind: StateKind.NoValue });
+
+			if (isObservable(initialComputation)) {
+				initialComputation
+					.subscribe({
+						next: (value) => sourceValue.set({ kind: StateKind.Value, value }),
+						error: (error) => sourceValue.set({ kind: StateKind.Error, error }),
+					})
+					// we need to unsubscribe because we don't want to keep the subscription
+					// we only care about the initial value
+					.unsubscribe();
+
+				if (sourceValue()!.kind === StateKind.NoValue)
```

Comment:
suggestion: no assertion needed
```suggestion
				if (sourceValue().kind === StateKind.NoValue)
```